spec/std/range_spec.cr

```diff
@@ -301,6 +356,49 @@ describe "Range" do
     it "is not empty with ... and begin.succ == end" do
       (1...2).reverse_each.to_a.should eq([1])
     end
+
+    it "raises on endless range" do
+      expect_raises(ArgumentError, "Can't reverse_each endless range") do
+        (1..(true ? nil : 1)).reverse_each
+      end
+    end
+  end
+
+  describe "step" do
```

Comment:
Can we test negative steps?

Reply:
Good idea, I totally forgot about this. I'll add it as an extra commit.

Reply:
Actually, at least in Ruby `step` can't be negative because a Range is always increasing, if it's decreasing (for example `5..1`) then it's "empty".

Let's discuss this in a separate issue, the current behavior is not yielding at all, but maybe giving an error would be better, or even making it work (using `pred`).

Reply:
Small correction: a negative step makes it hang. Definitely not good but I still think we should discuss it in a separate issue (I'll open one soon).

Reply:
:+1: